core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.spec.ts

```diff
@@ -626,6 +708,39 @@ describe('OptimizedSsrEngine', () => {
       engineRunner.request('a');
       expect(engineRunner.renders).toEqual(['', 'a-0']);
     }));
+
+    it('should fallback to CSR when forcedSsrTimeout is exceeded for a request', fakeAsync(() => {
+      const engineRunner = new TestEngineRunner(
+        {
+          renderingStrategyResolver: () => RenderingStrategy.ALWAYS_SSR,
+          timeout: 50,
+          forcedSsrTimeout: 150,
+        },
+        defaultRenderTime,
+        true
+      );
+      engineRunner.request('a');
+      tick(200);
+      expect(engineRunner.renders).toEqual([new Error('a-0')]);
+      flush();
+    }));
```

Comment:
1. IMHO the idea for the test is good, but the test body doesn't match it the description
We should:
- expect a CSR fallback: `toEqual([''])`
- forcedSsrTimeout should be e.g. 75 - to be lower than the configured rendering time 100=defaultRenderTime

2. 
> should fallback to CSR when forcedSsrTimeout is exceeded for a request, _regardless that the render eventually failed_